### PR TITLE
Implement new Style/ExponentialNotation cop

### PR DESCRIPTION
**Pull request by tdeo**
This cop implements the rule describe in the "exponential notation"
section of the ruby style guide.
It allows for an additional style not mentionned in the guide,
`integral` which allows only integer mantissa, without any
trailing zero.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/exponential_notation.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+# TODO: when finished, run `rake generate_cops_documentation` to update the docs
```
Comment (bbatsov) [inline lib/rubocop/cop/style/exponential_notation.rb:3]: You should remove this line. :-)

File: lib/rubocop/cop/style/exponential_notation.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+module RuboCop
+  module Cop
+    module Style
+      # This cop enforces consistency when using exponential notation
+      # for numbers in the code (eg 1.2e4). Multiple modes can be selected
```
Comment (bbatsov) [inline lib/rubocop/cop/style/exponential_notation.rb:8]: Multiple modes -> Different style are supported. 

That makes it consistent with RuboCop's terminology. I'd also add the explanations of the supported style before the examples - again, for consistency with other cops.

File: config/default.yml [hunk 3]
```diff
@@ -2740,6 +2740,18 @@ Style/ExpandPathArguments:
   Enabled: true
   VersionAdded: '0.53'
 
+Style/ExponentialNotation:
+  Description: 'When using exponential notation, favor a mantissa between 1 (inclusive) and 10 (exclusive).'
+  StyleGuide: '#exponential-notation'
+  Reference: 'https://github.com/rubocop-hq/ruby-style-guide/issues/803'
```
Comment (bbatsov) [inline config/default.yml:2746]: That's normally used for links to resources like wikipedia, blog posts, etc.

**Comment by tdeo** [comment]
Thanks for the review, I just amended the commit and pushed the branch

File: CHANGELOG.md [hunk 4]
```diff
@@ -30,6 +30,7 @@
 * [#7816](https://github.com/rubocop-hq/rubocop/pull/7816): Support Ruby 2.7's numbered parameter for `Style/Lambda`. ([@koic][])
 * [#7829](https://github.com/rubocop-hq/rubocop/issues/7829): Fix an error for `Style/OneLineConditional` when one of the branches contains `next` keyword. ([@koic][])
 * [#7384](https://github.com/rubocop-hq/rubocop/pull/7384): Add new `Style/DisableCopsWithinSourceCodeDirective` cop. ([@egze][])
+* Add a new `Style/ExponentialNotation` cop. ([@tdeo][])
```
Comment (bbatsov) [inline CHANGELOG.md:33]: Btw, you should move this to `master`, as 0.81 was already released.

**Comment by tdeo** [comment]
Amended and rebased again :)

**Comment by bbatsov** [comment]
Victory! 🎉 Thanks for working on this! 🙇‍♂️

**Comment by tdeo** [comment]
Thanks :tada:

**Pull request merged**